lib/fluent/winsvc.rb

```diff
@@ -57,8 +59,16 @@ def initialize(service_name)
     
     def service_main
       @pid = service_main_start(@service_name)
-      while running?
-        sleep 10
+      begin
+        loop do
+          sleep 5
+          break unless running?
+          pid, status = Process.waitpid2(@pid, Process::WNOHANG)
+          raise Errno::ECHILD if pid || status
```

Comment:
```suggestion
          raise Errno::ECHILD unless Process.waitpid(@pid, Process::WNOHANG)
```

If we don't use `status`, we can simplify the code.

Reply:
Done